programs/stake/src/stake_state.rs

```diff
@@ -842,6 +817,83 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
     }
 }
 
+/// After calling `validate_delegate_amount()`, this struct contains calculated values that are used
+/// by the caller.
+struct ValidatedDelegateInfo {
+    stake_amount: u64,
+}
+
+/// Ensure the stake delegation amount is valid.  This checks that the account meets the minimum
+/// balance requirements to delegate stake.  If not, return an error.
+fn validate_delegate_amount(
+    account: &KeyedAccount,
+    meta: &Meta,
+) -> Result<ValidatedDelegateInfo, InstructionError> {
+    let stake_amount = account.lamports()?.saturating_sub(meta.rent_exempt_reserve); // can't stake the rent ;)
+    Ok(ValidatedDelegateInfo { stake_amount })
+}
+
+/// After calling `validate_split_amount()`, this struct contains calculated values that are used
+/// by the caller.
+#[derive(Copy, Clone, Debug, Default)]
+struct ValidatedSplitInfo {
+    src_remaining_balance: u64,
+    dst_rent_exempt_reserve: u64,
+}
+
+/// Ensure the split amount is valid.  This checks the source and destination accounts meet the
+/// minimum balance requirements, which is the rent exempt reserve plus the minimum stake
+/// delegation, and that the source account has enough lamports for the request split amount.  If
+/// not, return an error.
+fn validate_split_amount(
+    src_account: &KeyedAccount,
+    dst_account: &KeyedAccount,
```

Comment:
Please spell out. Can probably drop _account suffix, though, if you're worried about length

Reply:
Done!